TraceScreen.c

```diff
@@ -131,6 +134,18 @@ bool TraceScreen_forkTracer(TraceScreen* this) {
 
 static void TraceScreen_updateTrace(InfoScreen* super) {
    TraceScreen* this = (TraceScreen*) super;
+
+   if (!this->status) {
+     fd_set fds_in;
+     FD_ZERO(&fds_in);
+     FD_SET(STDIN_FILENO, &fds_in);
+
+     struct timeval tv_in = { .tv_sec = 1, .tv_usec = 0 };
+     select(STDIN_FILENO + 1, &fds_in, NULL, NULL, &tv_in);
```

Comment:
This somewhat duplicates the code below that is used for regular setup for the `fd_set`.

I think, we can achieve the same effect by doing the following:
1. Uncomment the `FD_SET(STDIN_FILENO, &fds);` line below
2. Only set `FD_SET(fd_strace, &fds);` IFF `this->strace_active` is still set (cf. note re naming of `status`)
3. Do the `select` on `MAXIMUM(STDIN_FILENO, fd_strace) + 1` for the max fd argument.

Reply:
Proposed patch:

```diff
diff --git a/TraceScreen.c b/TraceScreen.c
index ce4db089..6e0be59c 100644
--- a/TraceScreen.c
+++ b/TraceScreen.c
@@ -135,30 +135,20 @@ err:
 static void TraceScreen_updateTrace(InfoScreen* super) {
    TraceScreen* this = (TraceScreen*) super;
 
-   if (!this->strace_alive) {
-     fd_set fds_in;
-     FD_ZERO(&fds_in);
-     FD_SET(STDIN_FILENO, &fds_in);
-
-     struct timeval tv_in = { .tv_sec = 1 };
-     select(STDIN_FILENO + 1, &fds_in, NULL, NULL, &tv_in);
-
-     return;
-   }
-
-   char buffer[1025];
-
    int fd_strace = fileno(this->strace);
-   assert(fd_strace != -1);
 
    fd_set fds;
    FD_ZERO(&fds);
-// FD_SET(STDIN_FILENO, &fds);
-   FD_SET(fd_strace, &fds);
+   FD_SET(STDIN_FILENO, &fds);
+   if (this->strace_alive) {
+      assert(fd_strace != -1);
+      FD_SET(fd_strace, &fds);
+   }
 
    struct timeval tv = { .tv_sec = 0, .tv_usec = 500 };
-   int ready = select(fd_strace + 1, &fds, NULL, NULL, &tv);
+   int ready = select(MAXIMUM(STDIN_FILENO, fd_strace) + 1, &fds, NULL, NULL, &tv);
 
+   char buffer[1025];
    size_t nread = 0;
    if (ready > 0 && FD_ISSET(fd_strace, &fds))
       nread = fread(buffer, 1, sizeof(buffer) - 1, this->strace);
```